sprint_7-solution

AlexViva3 opened the pull request:
Попытка номер 1!

Commit:
ТЗ 6 Добавил в ветку класс Node и TaskCycler, сделал первый код по двухсвязному списку и истории тасков :(

Commit:
:)

Commit:
:)

Commit:
:^)

Commit:
:^)

Commit:
:^)

Commit:
^-^

Commit:
^-^

Commit:
^-^

Commit:
^=^

Commit:
^=^

Commit:
^=^

Commit:
^-^

Commit:
Ненавижу проверку кодстайла :(

Commit:
Ненавижу проверку кодстайла >:(

Commit:
:)

Commit:
:)

avfyodorov reviewed (commented):
Добрый вечер, Александр! 

К сожалению, Вы прислали код от предыдущего домашнего задания. Проверьте, пожалуйста.

Commit:
:)

Commit:
Исправил все по checkstyle, надеюсь кайфую

Commit:
Исправил все по checkstyle, надеюсь кайфую

Commit:
Исправил все по checkstyle, надеюсь кайфую

Commit:
Исправил все по checkstyle, надеюсь кайфую

Commit:
Исправил все по checkstyle, надеюсь кайфую

Commit:
Исправил все по checkstyle, надеюсь кайфую

Commit:
Исправил все по checkstyle, надеюсь кайфую

avfyodorov commented on src/Main.java:
```diff
@@ -1,6 +1,5 @@
-import datapacks.SubEpicTusk;
 import datapacks.Task;
 import management.Managers;
 import management.TaskManager;
```
Все джар файлы лучше убрать из репозитария, они совершенно не нужны.

avfyodorov commented on src/management/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,143 @@
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            System.err.println("Ошибка сохранения: " + e.getMessage());
```
Тут немного   не так. Перехватываете IOException и сами пробрасываете
 throw new ManagerSaveException(«Ошибка и т.д.."); 

```java
  ...........  
      } catch (IOException e) {
            throw new ManagerSaveException("Ошибка при записи в  файл: " + e.getMessage());
        }
```

avfyodorov commented on src/management/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,143 @@
+                }
+            }
+        } catch (IOException e) {
+            System.err.println("Ошибка сохранения: " + e.getMessage());
```
И здесь аналогично
```java
  ...........  
      } catch (IOException e) {
            throw new ManagerSaveException("Ошибка при чтении файла: " + e.getMessage());
        }
```

avfyodorov commented on src/management/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,143 @@
+            }
+        } catch (IOException e) {
+            System.err.println("Ошибка сохранения: " + e.getMessage());
+            // Использую System.err для вывода ошибок
```
По ТЗ:
Исключения вида IOException нужно отлавливать внутри метода save и выкидывать собственное непроверяемое исключение ManagerSaveException. Благодаря этому можно не менять сигнатуру методов интерфейса менеджера. 
.-----------------

Собственное непроверяемое исключение должно выглядеть приблизительно так: 
```java
public class ManagerSaveException extends RuntimeException {
	public ManagerSaveException(String message, Exception cause) {
		super(message, cause);
	}

}
```

Лучше наследовать от RuntimeException
Из ТЗ:
кидать собственное непроверяемое исключение ManagerSaveException. Благодаря этому можно не менять сигнатуру методов интерфейса менеджера.

public class RuntimeException extends Exception — базовый класс для ошибок во время выполнения. Относится к необрабатываемым исключениям (uncatched\unchecked). Как сказано в описании класса — это суперкласс, исключения которого могут быть выброшены во время нормальной работы JVM.

avfyodorov commented on src/management/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,143 @@
+            // Использую System.err для вывода ошибок
+        }
+    }
+
```
Собственные исключения принято хранить в отдельном пакете, например, exceptions

avfyodorov commented on src/management/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,143 @@
+                task.getName(),
+                task.getStatus().name(),
+                task.getDescription(),
+                task instanceof SubEpicTusk ?
```
instanceof в ООП использовать не приветствуется. Лучше воспользоваться полиморфизмом и спросить у самого обьекта getType.
Есть же соответствующий метод

avfyodorov reviewed (commented):
Добрый день, Александр! 

Хорошая работа, от меня  несколько уточнений.

Commit:
Я думаю, все исправил :)
Я не особо понимаю как убрать драж файлы если они уже в самом репозитории, в гитигнор я добавил уже их :(
